docs/semgrep-code/remove-duplicates.md

```diff
@@ -3,29 +3,111 @@ slug: remove-duplicates
+:::info
+The calculations used to determine whether findings are new are subject to change at any time as Semgrep improves its deduplication logic.
+:::
+
+### `match_based_id`
+
+Semgrep generates the `match_based_id`, the primary fingerprint used to track and deduplicate findings, for a finding using the following information:
+
+- The file path
+- The name of the rule that generated the finding
+- The rule pattern with the metavariables' values substituted in
+
+This information is combined and then hashed. At this point, Semgrep appends the **index**, a value generated by determining the number of times the rule involved matched code in the file. Note that the index is appended to the hash, not combined with the other finding information before hashing. This is done to preserve information on how findings are related. For example, `finding0` with `match_based_id = 123_0` and `finding1` with `match_based_id = 123_1` indicate that both were generated from the same rule matching the same code pattern in the same file.
+
+Because Semgrep uses the rule pattern instead of the literal code syntax to generate `match_based_id`, code changes that don't impact the code pattern matched also don't hinder Semgrep's ability to recognize that the finding isn't a duplicate of an existing finding.
+
+![Semgrep AppSec Platform groups together findings on different branches](/img/matched-findings.png)
+_**Figure**. Semgrep AppSec Platform groups together the same finding identified as present on multiple branches._
+
+For example, if the original file scanned is:
+
+```python
+a = 1
+b = 2
+spcd.get("foo")
+c = 3
+d = 4
+sink("foo")
+```
+
+The rule pattern identified and used in generating the `match_based_id` is:
+
+```python
+spcd.get("foo")
+...
+sink("foo")
+```
+
+If the following change is made to the original file:
+
+```python
+a = 1
+b = 2
+spcd.get("foo")
+c = 3
+c_1 = 5
+d = 4
+sink("foo")
+```
+
+The rule pattern identified and used in generating the `match_based_id` does not change:
+
+```python
+spcd.get("foo")
+...
+sink("foo")
+```
+
+This means that the `match_based_id` itself doesn't change, allowing Semgrep to identify that the two findings are the same, allowing deduplication. Furthermore, this process enables Semgrep to ignore lines that do not impact code function. For example, adding comments to a code file would create a new `syntactic_id`, but not a new `match_based_id`.
+
+### `syntactic_id`
+
+Semgrep generates the `syntactic_id` for a finding using the following information:
+
+- The file path
+- The name of the rule that generated the finding
+- The code syntax, or the literal piece of code that matched the rule
+- The index, a value generated by determining the number of times the rule involved matched code in the file
+
+This information is combined and then hashed before being stored for privacy.
```

Comment:
```suggestion
This information is combined and then hashed for privacy before being stored.
```

"being stored for privacy" - the modifier "for privacy" was too far from the "hash", it could be interpreted two ways (hashed for privacy, or stored for privacy) at first read, so I am suggesting to put "for privacy" nearer to "hash".